internal/service/news/news.go

```diff
@@ -0,0 +1,123 @@
+package news
+
+import (
+	gocontext "context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/newUser1337/task-news/internal/config"
+	"github.com/newUser1337/task-news/internal/context"
+	"github.com/newUser1337/task-news/internal/entity"
+	"github.com/newUser1337/task-news/internal/httpclient"
+	"github.com/newUser1337/task-news/internal/repository"
+)
+
+const articlesListEndpoint string = "https://www.htafc.com/api/incrowd/getnewlistinformation?count=50"
+const articleEndpoint string = "https://www.htafc.com/api/incrowd/getnewsarticleinformation?id=%d"
+
+type articleRepository interface {
+	SaveArticleList(gocontext.Context, []*entity.NewsletterNewsItemDB) ([]int64, error)
+	SaveArticle(gocontext.Context, *entity.NewsArticleDB) (int64, error)
+}
+
+type NewsServer struct {
+	appCtx  *context.AppCtx
+	article articleRepository
+}
+
+func StartNewNewsServer(appCtx *context.AppCtx, cfg *config.Config, rep *repository.Repository) {
+	newsServer := &NewsServer{
+		appCtx:  appCtx,
+		article: rep.Article,
+	}
+
+	go newsServer.maintain(cfg)
+}
+
+func (n *NewsServer) maintain(cfg *config.Config) {
```

Comment:
config is usually not supposed to be updated, so better pass it by value (or pass refresh interval only as you don't need anything else)

Reply:
fixed